apollo-federation/src/query_plan/operation.rs

```diff
@@ -701,6 +701,10 @@ impl Selection {
         inline_fragment: InlineFragment,
         sub_selections: SelectionSet,
     ) -> Self {
+        assert_eq!(
```

Comment:
we should maybe keep this as a debug_assert so it doesn't bring down any routers.
```suggestion
        debug_assert_eq!(
```
That or making the function return a Result

Reply:
Good point!